src/ring_theory/int/basic.lean

```diff
@@ -341,3 +341,62 @@ instance decidable_int : decidable_rel (λ a b : ℤ, (multiplicity a b).dom) :=
 λ a b, decidable_of_iff _ finite_int_iff.symm
 
 end multiplicity
+
+lemma induction_on_primes [P: nat → Prop] (h₁: P 0) (h₂: P 1)
+  (h: ∀ p a: ℕ, nat.prime p → P a → P (p * a)): ∀ n: ℕ, P n :=
+begin
+  intro n,
+  apply unique_factorization_monoid.induction_on_prime,
+  exact h₁,
+  { intros n h,
+    rw nat.is_unit_iff.1 h,
+    exact h₂, },
+  { intros a p _ hp ha,
+    exact h p a (nat.prime_iff_prime.2 hp) ha, },
+end
+
+lemma int.associated_nat_abs (k: ℤ): associated k k.nat_abs :=
+begin
+  apply associated.symm,
+  unfold associated,
+  cases int.nat_abs_eq k with h h,
+  use 1, simp, exact h.symm,
+  use -1, simp, exact h.symm,
+end
+
+lemma int.prime_iff_nat_abs_prime {k: ℤ}: prime k ↔ nat.prime k.nat_abs :=
+begin
+  rw nat.prime_iff_prime_int,
+  rw prime_iff_of_associated (int.associated_nat_abs k),
+end
+
+lemma int.associated_iff (a: ℤ) (b: ℤ): associated a b ↔ (a = b ∨ a = -b) :=
```

Comment:
This would be easier to prove with these lemmas:

```lean
lemma int.nat_abs_eq_nat_abs_iff {a b : ℤ} : a.nat_abs = b.nat_abs ↔ a = b ∨ a = -b :=
begin
  split,
  { intro h,
    apply_fun coe at h,
    cases int.nat_abs_eq a with ha ha;
    cases int.nat_abs_eq b with hb hb;
    rw [ha, hb],
    { left, exact h },
    { right, rwa neg_neg },
    { right, rw h },
    { left, rw h } },
  { rintro (rfl|rfl),
    { refl },
    { exact int.nat_abs_neg b } },
end

theorem int.associated_iff_nat_abs {a b : ℤ} : associated a b ↔ a.nat_abs = b.nat_abs :=
begin
  rw [←dvd_dvd_iff_associated, ←int.nat_abs_dvd_abs_iff, ←int.nat_abs_dvd_abs_iff, dvd_dvd_iff_associated],
  exact associated_iff_eq,
end
```

Reply:
Easier indeed, thanks.